### PR TITLE
Switch to Validictory, initial stages of validation tests

### PR DESCRIPTION
**Pull request by klsmith-usgs**

### CODE REVIEW

**Comment by caustin-usgs** [comment]
does the validictory module to requirements.txt ?

**Comment by klsmith-usgs** [comment]
Yes it does, I'll get it added.

File: api/providers/validation/__init__.py [hunk 1]
```diff
@@ -1,2 +1,84 @@
...
+    def fetch_order_schema(self):
+        """Return the validation schema"""
+
+    @abc.abstractmethod
+    def __call__(self):
+        pass
+
+
+class MockValidationProvider(ValidationInterfaceV0):
+    def validate(self, order, username):
```
Comment (caustin-usgs) [inline api/providers/validation/__init__.py]: where does the MockValidationProvider class get used?

Comment (klsmith-usgs) [inline api/providers/validation/__init__.py]: Legacy item? We have the mock providers grouped under api.ordering.providers

**Comment by caustin-usgs** [comment]
looks good to me, go ahead and merge if you're ready

**Pull request merged**